python/ssm-compliance-check-lambda-fn.py

```diff
@@ -0,0 +1,110 @@
+#{
+#    "Version": "2012-10-17",
+#    "Statement": [
+#        {
+#            "Effect": "Allow",
+#           "Action": [
+#                "logs:CreateLogGroup",
+#                "logs:CreateLogStream",
+#                "logs:PutLogEvents"
+#            ],
+#            "Resource": "arn:aws:logs:*:*:*"
+#        },
+#        {
+#            "Effect": "Allow",
+#            "Action": [
+#                "config:Put*",
+#                "config:Get*",
+#                "config:List*",
+#                "config:Describe*",
+#                "ec2:Describe*",
+#                "ssm:Describe*"
+#            ],
+#            "Resource": "*"
+#        }
+#    ]
+#}                                                                                     
+################################################################################################################
+
+import boto3
+import json
+
+def evaluate_compliance(instanceid):
+    ssmclient = boto3.client('ssm')
+    ssmstatus = ssmclient.describe_instance_information(
+        InstanceInformationFilterList=[
+            {
+                'key': 'InstanceIds',
+                'valueSet': [str(instanceid)]
+            }
+        ]
+    )
+    if ssmstatus['InstanceInformationList']:
+        flag = ssmstatus['InstanceInformationList']
+        for version in flag:
+            if version['IsLatestVersion']:
+                return {
+                        'compliance_type': "COMPLIANT",
+                        'annotation': "The instance is running latest ssm agent version"
+                       }
+            else:
+                return {
+                        'compliance_type': "NON_COMPLIANT",
+                        'annotation': "The instance is running an outdated ssm agent version"
+                       }
+    else:
+        return {
+                'compliance_type': "NON_COMPLIANT",
+                'annotation': "SSM agent is not installed"
+               }
+                   
+def evaluation_result(instanceid):
+    ec2client = boto3.client('ec2')
+    instancedetail = ec2client.describe_instance_status(InstanceIds=[str(instanceid)])
+    if instancedetail['InstanceStatuses']:
+        response = evaluate_compliance(instanceid)
+        return response
+    else:
+        return {
+                'compliance_type' : "NOT_APPLICABLE",
+                'annotation' : "Instance is not in running state"
+               }
+
+def lambda_handler(event, context):
+    print(event)
+    invoking_event = json.loads(event['invokingEvent'])
+    configclient = boto3.client('config')
+    respond = configclient.list_discovered_resources(resourceType='AWS::EC2::Instance')
+    print(respond)
+    for resource in respond['resourceIdentifiers']:
+            instanceid = resource['resourceId']
+            evaluationresult = evaluation_result(instanceid) 
+            print(instanceid, evaluationresult)
+            response = configclient.put_evaluations(

```

Comment:
You could collect all the evaluations and then make one PutEvaluations API call (or batch them up into 100's)